# HG changeset patch
# User Andrew Cooper <andrew.cooper3@citrix.com>
# Date 1394717824 -3600
# Node ID 4cd93b5a74411cc0c992842aa68bec905fc27358
# Parent  40b60c5823ea9eb5c0d4c2400f650139ad7c34d9
x86/time: initialise time earlier during start_secondary()

It is safe to do so, and useful for "[second.microseconds]" style timestamps
on printk()s during secondary bringup.

Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>
Acked-by: Keir Fraser <keir@xen.org>

diff -r 40b60c5823ea -r 4cd93b5a7441 xen/arch/x86/smpboot.c
--- a/xen/arch/x86/smpboot.c
+++ b/xen/arch/x86/smpboot.c
@@ -347,6 +347,8 @@ void start_secondary(void *unused)
 
     percpu_traps_init();
 
+    init_percpu_time();
+
     cpu_init();
 
     smp_callin();
@@ -381,8 +383,6 @@ void start_secondary(void *unused)
     cpumask_set_cpu(cpu, &cpu_online_map);
     unlock_vector_lock();
 
-    init_percpu_time();
-
     /* We can take interrupts now: we're officially "up". */
     local_irq_enable();
     mtrr_ap_init();
